src/sources/tcp.rs

```diff
@@ -1,11 +1,7 @@
 use super::util::{SocketListenAddr, TcpSource};
```

Comment:
I'm not sure if it makes sense to leave tcp.rs and udp.rs in the sources/ dir now that they're no longer top-level sources but really just helper functionality for the socket source.  If there's a better place for these files please LMK!

Reply:
Yeah, there's not a lot left here and I agree it makes sense to merge them into `socket.rs`. If we find that it's getting too big, it's simple enough to split `socket.rs` in `socket/{mod,tcp,udp}.rs`.

Reply:
I went with adding a socket dir and making udp and tcp sub-modules of socket as I thought it was still nice to keep most of this functionality separated rather than all mixed into one large socket.rs file.  Please let me know what you think.